pkg/v1/tkg/tkgctl/create_cluster.go

```diff
@@ -305,9 +306,9 @@ func (t *tkgctl) configureCreateClusterOptionsFromConfigFile(cc *CreateClusterOp
 
 	// set BuildEdition from config variable
 	if cc.Edition == "" {
-		edition, err := t.TKGConfigReaderWriter().Get(constants.ConfigVariableBuildEdition)
+		cfg, err := config.GetClientConfig()
```

Comment:
We should try not to get an edition from config file at this place as this is part of the library code and we might want the library users to provide an edition with the `BUILD_EDITION` config variable or through the library interface.

If we think we don't want to allow users to specify `BUILD_EDITION` config variable then we need to communicate with library consumers to always pass `edition` as part of the `CreateClusterOptions` or `InitRegionOptions` argument.

Reply:
@anujc25 , we'll continue to allow the BUILD_EDITION flag in the cluster config file as before (no change).